aten/src/ATen/SparseTensorImpl.cpp

```diff
@@ -0,0 +1,70 @@
+#include <ATen/ATen.h>
+#include <ATen/SparseTensorImpl.h>
+
+namespace at {
+
+// SparseTensorImpl defaults to a [0] size tensor with one sparse dimension and
```

Comment:
I find this comment a little confusing; can you structure it as:
- This is what it _should_ be (i.e.size [1,0])
- Since we don't have support for that, this is what we chose to do instead

Reply:
```
// An empty dense tensor defaults to a 1-dimensional tensor of size [0]
// (recall, it is not a 0-dimensional tensor, because such a tensor would
// a scalar and have one element)
//
// Thus, an empty sparse tensor should be a 1-dimensional tensor of size [0].
// Furthermore, we have dim == sparseDims + denseDims; since this is a sparse
// tensor, let us say that an empty sparse tensor has sparseDims == 1 and
// denseDims == 0.  (There is a degree of freedom here, but given that this
// is a sparse dimension, it seems reasonable to demand that sparseDims > 0).
//
// In an ideal world, this would then mean we allocate a [1,0] size indices
// tensor and a [0] size values tensor for such an empty tensor.  However,
// we don't currently support zero-size dimensions, so we can't actually
// do this; so we just allocate zero-size tensors for everything.
```

Actually, there is a bit in the comment defending why we should default sparseDims == 1, denseDims == 0, I reworded it for clarity.